rollup/circuitcapacitychecker/impl.go

```diff
@@ -0,0 +1,117 @@
+//go:build circuit_capacity_checker
+
+package circuitcapacitychecker
+
+/*
+#cgo LDFLAGS: -lm -ldl -lzkp -lzktrie
+#include <stdlib.h>
+#include "./libzkp/libzkp.h"
+*/
+import "C" //nolint:typecheck
+
+import (
+	"encoding/json"
+	"sync"
+	"unsafe"
+
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/log"
+)
+
+func init() {
+	C.init()
+}
+
+type CircuitCapacityChecker struct {
+	sync.Mutex
+	id uint64
+}
+
+func NewCircuitCapacityChecker() *CircuitCapacityChecker {
+	id := C.new_circuit_capacity_checker()
+	return &CircuitCapacityChecker{id: uint64(id)}
+}
+
+func (ccc *CircuitCapacityChecker) Reset() {
+	ccc.Lock()
+	defer ccc.Unlock()
+
+	C.reset_circuit_capacity_checker(C.uint64_t(ccc.id))
+}
+
+func (ccc *CircuitCapacityChecker) ApplyTransaction(traces *types.BlockTrace) (*types.RowConsumption, error) {
+	ccc.Lock()
+	defer ccc.Unlock()
+
+	tracesByt, err := json.Marshal(traces)
+	if err != nil {
+		return nil, ErrUnknown
+	}
+
+	tracesStr := C.CString(string(tracesByt))
+	defer func() {
+		C.free(unsafe.Pointer(tracesStr))
+	}()
+
+	log.Info("start to check circuit capacity for tx")
```

Comment:
Are these logs added temporarily for testing?

Reply:
I thought we wanted to add them. but I can remove

Reply:
Maybe debug logs. But we need to add more context to make them useful.

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/pull/409